### ISSUE TITLE
[doc] Referencing GitHub issues?

### ISSUE THREAD

**Issue by Grinnz** [Perl/perl5#17331]
Currently, https://github.com/Perl/perl5/blob/blead/Porting/how_to_write_a_perldelta.pod#style says that rt.perl.org tickets should be referenced like `perl #43010` or `[perl #43010]`. Similarly, references exist throughout the source code in comments and documentation in that style as well as things like `[rt.perl.org #43010]`, `rt #43010`, `RT #43010`, or omitting the space or hash. What style should be recommended for referencing GitHub issues, and what should be used when converting these existing references to the corresponding GitHub issue number?

My suggestion is `[github #43010]` and `GH #43010` accordingly; reusing `[perl #43010]` while strictly accurate would unfortunately be ambiguous with RT ticket numbers.

**Comment by Grinnz** [comment]
In https://perldoc.pl/5.31.6/perldelta issues were referenced simply as [#43010]. This also seems too ambiguous.

**Comment by khwilliamson** [comment]
I'm fine with this change

**Comment by tonycoz** [comment]
Yeah, the bare ids `[#xxxx]` is mostly my fault, I think I did all of them in the 5.31.6 delta except one, and Karl probably followed my example.

I'm not really fond of any of the possibilities, `[github #xxxx]` or `[gh #xxxx]` implies we're talking about issues in github itself.

Right now I think `[#xxxx]` is the best possiblity, it's distinct from `[perl #xxxx]`, and the 5.31.6 delta mentions we've moved to github.

Mentions in code, including tests should probably refer to gh or github issues.

I can see it being a problem for @xsawyerx for the 5.32 release, he might choose to leave the rt references in 5.31.[0-5] as `[perl #xxxx]` for the 5.32.0 delta or translate them all to github references.

**Comment by Grinnz** [comment]
I agree it's not ideal, but I think it's well established that "GH #XXXXX" refers to a github issue for whatever project you happen to be discussing, given the widespread nature.

**Comment by Grinnz** [comment]
Just for prior art, though of course none of these are in the situation of having to disambiguate from a prior tracker:

* Elasticsearch uses bare issue numbers (https://www.elastic.co/guide/en/elasticsearch/reference/current/release-notes-7.0.0.html)
* VSCode uses bare issue numbers, and prefix PR for pull requests (https://code.visualstudio.com/updates/v1_40)
* PHP uses bare issue numbers (https://www.php.net/ChangeLog-7.php)

**Comment by Grinnz** [comment]
Regarding old perldelta references, I did ponder in https://github.com/Perl/perl5/pull/17329 whether it's worth updating *all* of them, because going forward people will know that github is the bugtracker and won't be as likely to know what RT is, when looking at old perldeltas. On the other hand, they are all linked pretty consistently.

**Comment by xsawyerx** [comment]
IMO, to make issues as locatable as possible publicly, we should use [GH #xxx], to indicate "this is the location where this issue is discussed."

I'm fine providing the GH issue ID even when it was moved from RT (meaning the [RT #xxx] still works, but I'm also okay with leaving the RT number for those - in case anyone has an opinion about it.

I don't see a need for both [github #xxx] and GH #xxx. We should normalize that. My preference is the latter.

**Comment by tonycoz** [comment]
Prompted by #18453 I went to document this in `Porting/how_to_write_a_perldelta.pod` but the discussion here doesn't cover some cases, I'm suggesting:

```
=item github.com

C<GH #43010> inline for perl issues, but enclose in square brackets after a sentence.
C<[GH #43010]>.  This is for consistency with historical practice for RT issues.

C<GH site/project #43010> for non-perl issues, but enclose in square brackets after
a sentence.  C<[GH site/project #43010]>.
```

Grinnz claimed PHP uses bare issue numbers, but as far as I can see it always prefixes that by "bug", so "bug #43010"

Elasticsearch uses "(issues: #29881, #37301)" in the cases where we've done "[perl #43010]"

**Comment by Grinnz** [comment]
For non-perl project issues that are on github it could be done consistent with github's own link formatting: `[GH project#43010]`

**Comment by kozachkov** [comment]
What about mentions of rt.perl.org notes in perl code now (5.33.9)? Will they be replaced by new GitHub issues links?

**Comment by Leont** [comment]
> What about mentions of rt.perl.org notes in perl code now (5.33.9)? Will they be replaced by new GitHub issues links?

Yes, that does sound helpful

**Comment by Grinnz** [comment]
I started on this some time ago but it dropped down my priority list, I'll see if the branch is still usable.
